File: data-structures/reverse-string.js

function reverse(str) {
    // check the value
    if (!str || str.length < 2 || typeof str !== 'string') {
        return 'Hey baby. Something is bad.';
    }

    const backwards = [];
    const totalItems = str.length - 1;
    for (let index = totalItems; index >= 0; index--) {
        backwards.push(str[index]);
    }

    return backwards.join('');
}

function reverse2(str) {
    return str.split('').reverse().join('');
}

const reverse3 = str => [...str].reverse().join('');

// const myString = 'Hola, mi nombre es Celso';
const myString = 12312;

console.log(reverse(myString));

File: data-structures/python/Array/reverse-string.py

def reverse(str):
    if not str:
        return None
    return str[::-1]

def reverse2(str):
    reversedWord = ''
    for i in range(len(str)):
        reversedWord = str[i] + reversedWord
    return "".join(reversedWord)

def reverse3(str):
    if (len(str) == 1):
        return str
    
    return str[len(str) - 1] + reverse3(str[:-1])

print(reverse(''))
print(reverse('Hola, mi nombre es Celso'))
print(reverse2('Hola, mi nombre es Celso'))
print(reverse3('Hola, mi nombre es Celso'))